buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -51,7 +51,7 @@ enum class ModuleType(val compatibility: JavaVersion) {
     WORKER(JavaVersion.VERSION_1_6),
     CORE(JavaVersion.VERSION_1_8),
     PLUGIN(JavaVersion.VERSION_1_8),
-    INTERNAL(JavaVersion.VERSION_1_7), //TODO bumpt to 8 after Groovy 2.5 upgrade
+    INTERNAL(JavaVersion.VERSION_1_8),
     REQUIRES_JAVA_8(JavaVersion.VERSION_1_8)
```

Comment:
Can we remove `REQUIRES_JAVA_8` now and use `INTERNAL` there, too?

Reply:
`REQUIRES_JAVA_8` is used exactly once, for the `testingJunitPlatform` subproject. The other `testing` subprojects use `ModuleType.WORKER` but the JUnit Platform requires Java 8. So it's not an internal project but we could rename it to `WORKER_JAVA_8` or sth. similar. Thoughts?

Reply:
@wolfs WDYT?

Reply:
What exactly is an internal project? Is worker == runs on worker? Plugin is a bundled plugin I guess. I also don't understand the difference between core and internal.

IOW: I don't understand the enum names here. So either we add some Javadoc or we rename them (I guess we should do both). And then we can come up with an answer to your question.

WDYT?

Reply:
If I interpreted it correctly, `INTERNAL` means that it will not be part of this distribution, e.g. `integTest`; `WORKER` means "parts of this subproject have to run in worker processes".

Would you be ok with addressing the renaming/documentation in a separate issue?

Reply:
Got it, now I understand what `INTERNAL` means.